controllers/auth.js

```diff
@@ -0,0 +1,45 @@
+const User = require('../models/User');
+const {StatusCodes} = require('http-status-codes');
+const {BadRequestError, UnauthenticatedError}=require('../errors');
+
+//to register a new user
+const register = async (req, res) => {
+    const user = await User.create({...req.body});
```

Comment:
Nitpick:
This doesn't need to be spread out because `req.body` is already an object.
```diff
- const user = await User.create({ ...req.body });
+ const user = await User.create(req.body);
```